### ISSUE TITLE
crashing before visualizing "complex" autogenerated forms for remote postgis layers

### ISSUE THREAD

**Issue by paolomartinis** [qgis/QGIS#47166]
### What is the bug or the crash?

QGIS crashes when trying to visualize "complex" forms generated using drag and drop designer for "remote" PostGIS layers.
For "complex" I'm meaning a form organized in tabs and groups with repeated widgets (the widget for a field in two different tabs).
The crash happens only with "remote" PostGIS layers, not with the same layer in a "local" PostGIS database (in the same machine where I'm running QGIS).
The crash happens only with already populated layers when clicking _Identify features_ or _Open Attribute Table_, not when inserting a new item
Removing one tab results in not crashing anymore.

### Steps to reproduce the issue

I think it's too difficult to reproduce the error without the possibility to connect to a "remote" PostGIS layer.
Maybe I can create a connectable PostGIS layer just for testing and provide the qml file?

### Versions

QGIS version
3.16.16-Hannover
QGIS code revision
f5778a89
Compiled against Qt
5.15.2
Running against Qt
5.15.2
Compiled against GDAL/OGR
3.4.1
Running against GDAL/OGR
3.4.1
Compiled against GEOS
3.10.0-CAPI-1.16.0
Running against GEOS
3.10.2-CAPI-1.16.0
Compiled against SQLite
3.35.2
Running against SQLite
3.35.2
PostgreSQL Client Version
13.0
SpatiaLite Version
5.0.1
QWT Version
6.1.3
QScintilla2 Version
2.11.5
Compiled against PROJ
8.2.1
Running against PROJ
Rel. 8.2.1, January 1st, 2022
OS Version
Windows 10 Version 2009
Active python plugins
AnotherDXF2Shape; 
Cxf_in; 
db-style-manager; 
ImportEpanetInpFiles; 
joinmultiplelines; 
MultiEdit; 
multiple_layers_tools; 
nominatim; 
ntv2_transformations; 
qepanet; 
QuickMultiAttributeEdit3; 
scriptrunner3; 
SpreadsheetLayers; 
wnt; 
db_manager; 
MetaSearch; 
processing

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by gioman** [comment]
> Maybe I can create a connectable PostGIS layer just for testing and provide the qml file?

@paolomartinis yes thanks. If you want you can send me the data privately.

But please also:

test on 3.22.3

and test on a new QGIS profile, no 3rd party plugins installed.

**Comment by paolomartinis** [comment]
Tested with 3.22.3 and new profile -> same results
I'll send you the connection parameters via PM, thanks

**Comment by gioman** [comment]
> I'll send you the connection parameters via PM, thanks

@paolomartinis thanks for having sent me the connection data and the QML file: I cannot replicate. Not by identifying feature, not by accessing the features form from the table of attributes. QGIS 3.22.3 on Windows10 and Ubuntu 20.04.

**Comment by gioman** [comment]
@paolomartinis I wonder is this shouldn't be reported to Microsoft, rather than here.

**Comment by paolomartinis** [comment]
@gioman
I'm not so sure if it's just for win11, now I tried with a fresh Ubuntu 20.04 virtual machine and QGIS 3.22 and it crashes already...
Tomorrow I'll try with a fresh win10 virtual machine.
But removing only the two calendar widgets in the first tab (leaving them in the second one) seems to solve the problem, at least for the linear layer I sent to you... Really don't know what's going on...

**Comment by gioman** [comment]
> now I tried with a fresh Ubuntu 20.04 virtual machine and QGIS 3.22 and it crashes the same way...

@paolomartinis I can also try it. I'll let you know.

**Comment by gioman** [comment]
> @paolomartinis I can also try it. I'll let you know.

sorry, already did it :)

**Comment by paolomartinis** [comment]
@gioman
I confirm, it's crashing also in a fresh VMware virtual machine with win10-21H2.
In both win11 and win10, QGIS is installed using osgeo4w network installation, choosing both qgis-full and qgis-ltr-full distributions either from download.osgeo.org and www.norbit.de mirrors. (I tried a lot of combinations :-) )
In ubuntu (running on a virtual machine) I installed only the qgis-ltr distribution.
The workaround of removing the calendar widgets is still working, so for now I'm ok with my project.
But I I hate "blind" crashes and dont't know if it can be an important bug, maybe it will come out with other issues related to visualization and/or PostGIS connections...
Really thanks, let me know if it's better to close the issue (and to close the testing connection).
